modules/account-lib/src/coin/hbar/tokenTransferBuilder.ts

```diff
@@ -0,0 +1,125 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import * as Long from 'long';
+import { TokenId } from '@hashgraph/sdk';
+import * as proto from '@hashgraph/proto';
+import { BuildTransactionError, InvalidParameterValueError } from '@bitgo/sdk-core';
+import { Recipient } from './iface';
+import { TransferBuilder } from './transferBuilder';
+import { Transaction } from './transaction';
+import {
+  getHbarTokenFromTokenName,
+  getHbarTokenFromId,
+  isTokenTransfer,
+  stringifyAccountId,
+  stringifyTokenId,
+} from './utils';
+import { BigNumber } from 'bignumber.js';
+
+export class TokenTransferBuilder extends TransferBuilder {
+  private _tokenName; // currently only support 1 token/transfer
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.tokenTransfers = this.buildTokenTransferData(); // set to list by the contract
+    return await super.buildImplementation();
+  }
+
+  private buildTokenTransferData(): proto.ITokenTransferList[] {
+    let tokenTransferAmount = new BigNumber(0); // total send amount for each token
+    const tokenId = getHbarTokenFromTokenName(this._tokenName)?.tokenId;
+    const tokenTransferData: proto.IAccountAmount[] = [
+      {
+        accountID: this.buildAccountData(this._source.address),
+        amount: Long.fromInt(0),
+      },
+    ];
+
+    this._recipients.map((recipient) => {
+      tokenTransferAmount = tokenTransferAmount.plus(recipient.amount);
+      tokenTransferData.push(
+        { accountID: this.buildAccountData(recipient.address), amount: Long.fromString(recipient.amount) }, // recipient
+      );
+    });
+    tokenTransferData[0].amount = Long.fromString(tokenTransferAmount.toString()).negate(); // update sender send amount
+
+    return [
+      {
+        token: this.buildTokenId(tokenId!),
```

Comment:
`tokenId` is optional and may be undefined

Reply:
This is just the result from `getHbarTokenFromName` which can return `undefined` while I already checked if `_tokenName` is a valid token when adding it, so it will always exist there during building